lib/charms/grafana_cloud_integrator/v0/cloud_config_requirer.py

```diff
@@ -96,52 +88,49 @@ def _events(self):
 
     @property
     def credentials(self):
-        return Credentials(
-            self._data.get("username", ""),
-            self._data.get("password", "")
-        )
+        if not all(
+            self._is_not_empty(x)
+            for x in [
+                self._data.get("username", ""),
+                self._data.get("password", ""),
+            ]):
+                return Credentials(
+                    self._data.get("username", ""),
+                    self._data.get("password", "")
+                )
+        return None
 
     @property
     def loki_ready(self):
-        return (
-            self._is_not_empty(self.credentials.username)
-            and self._is_not_empty(self.credentials.password)
-            and self._is_not_empty(self.loki_url))
+        return self._is_not_empty(self.loki_url)
 
     @property
     def loki_endpoint(self) -> dict:
         """Return the loki endpoint dict."""
         if not self.loki_ready:
             return {}
 
-        return {
-            "url": self.loki_url,
-            "basic_auth": {
-                "username": self.credentials.username,
-                "password": self.credentials.password,
-            },
-        }
+        endpoint = {}
+        endpoint["url"] = self.loki_url
+        if self.credentials:
```

Comment:
`bool(self.credentials)` is guaranteed to be true, no? or does it override `__bool__`?

Reply:
It was before this change; the purpose it not make credentials compulsory anymore, so that the charm can work with endpoints that don't have basic authentication